tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -189,9 +197,57 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
             offset += WriteTags(ref bytes, offset, in spanModel, tagProcessors);
             offset += WriteMetrics(ref bytes, offset, in spanModel, tagProcessors);
 
+            if (metaStructValues?.Count > 0)
+            {
+                WriteMetaStruct(ref bytes, ref offset, metaStructValues);
+            }
+
```

Comment:
I agree with @lucaspimentel, there's already an style on how we write Tags and Metrics, we should treat `MetaStruct` in the same level and handle it in the same way. That style is a result of several iterations to make it the most performant possible. I wouldn't reinvent the wheel in this case.